src/System.Windows.Forms/src/System/Windows/Forms/LabelEditAccessibleObject.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Windows.Win32.System.Com;
+using Windows.Win32.UI.Accessibility;
+using UiaCore = Interop.UiaCore;
+
+namespace System.Windows.Forms;
+
+internal unsafe class LabelEditAccessibleObject : AccessibleObject
+{
+    private readonly WeakReference<LabelEditNativeWindow> _labelEdit;
+    private readonly LabelEditUiaTextProvider _textProvider;
+    private int[]? _runtimeId;
+
+    public LabelEditAccessibleObject(Control owningControl, LabelEditNativeWindow labelEdit)
+    {
+        ArgumentNullException.ThrowIfNull(owningControl);
+
+        _labelEdit = new(labelEdit);
+        UseStdAccessibleObjects(labelEdit.Handle);
+        _textProvider = new(owningControl, labelEdit, this);
+    }
+
+    internal override UiaCore.IRawElementProviderFragment? FragmentNavigate(UiaCore.NavigateDirection direction)
+        => direction switch
+        {
+            UiaCore.NavigateDirection.Parent => Parent,
+            _ => base.FragmentNavigate(direction)
+        };
+
+    internal override object? GetPropertyValue(UIA_PROPERTY_ID propertyID)
+        => propertyID switch
+        {
+            UIA_PROPERTY_ID.UIA_ProcessIdPropertyId => Environment.ProcessId,
+            UIA_PROPERTY_ID.UIA_ControlTypePropertyId => UIA_CONTROLTYPE_ID.UIA_EditControlTypeId,
+            UIA_PROPERTY_ID.UIA_AccessKeyPropertyId => string.Empty,
+            UIA_PROPERTY_ID.UIA_HasKeyboardFocusPropertyId => true,
+            UIA_PROPERTY_ID.UIA_IsKeyboardFocusablePropertyId => (State & AccessibleStates.Focusable) == AccessibleStates.Focusable,
```

Comment:
We have HasFlags() function for this, which is a bit more readable
```suggestion
            UIA_PROPERTY_ID.UIA_IsKeyboardFocusablePropertyId => State.HasFlags(AccessibleStates.Focusable),
```

Reply:
Done.